### PR TITLE
Add support for the demiurge keyboard

### PR DESCRIPTION
**Pull request by ojthetiny**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/demiurge/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Demiurge", 
+    "url": "", 
+    "maintainer": "ojthetiny", 
+    "width": 17,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/demiurge/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/demiurge/config.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#define RGBLED_NUM 21
+#define RGBLIGHT_ANIMATIONS
+#define RGBLIGHT_LIMIT_VAL 153
+#endif
+
+
+
+
+
+
```
Comment (zvecr) [inline keyboards/demiurge/config.h:62]:
```suggestion
```

File: keyboards/demiurge/config.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (zvecr) [inline keyboards/demiurge/config.h:1]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/demiurge/config.h [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 1
+#endif
+
+#define RGB_DI_PIN F1
+#ifdef RGB_DI_PIN
+#define RGBLED_NUM 21
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/demiurge/config.h:53]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/demiurge/info.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Demiurge", 
+    "url": "", 
+    "maintainer": "ojthetiny", 
+    "width": 17,
+    "height": 5,
```
Comment (drashna) [inline keyboards/demiurge/info.json:6]:
```suggestion
```

File: keyboards/demiurge/readme.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+![demiurge-img-1](https://imgur.com/a/mCoEfMa.png)
+
+60% XT
+
+- Keyboard Maintainer: [ojthetiny](https://github.com/ojthetiny)
+- Hardware Supported: Demiurge PCB
+- Hardware Availability: Through group-buy
+
+Make examples for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/demiurge/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/demiurge/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = yes	    # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no	    # Mouse keys
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes		    # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/demiurge/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes		    # Enable N-key Rollover
```

File: keyboards/demiurge/rules.mk [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	    # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/demiurge/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/demiurge/readme.md (hunk 6 above)
Comment (ojthetiny) [reply keyboards/demiurge/readme.md:12]: added bootloader info

**Review by drashna — APPROVED** [review]

File: keyboards/demiurge/rules.mk [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no	    # Mouse keys
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes		    # Enable N-key Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/demiurge/rules.mk:19]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/demiurge/rules.mk [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
```
Comment (fauxpark) [inline keyboards/demiurge/rules.mk:8]:
```suggestion
#   change yes to no to disable
```

File: keyboards/demiurge/readme.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
+# Demiurge
+
+![demiurge-img-1](https://imgur.com/a/mCoEfMa.png)
```
Comment (fauxpark) [inline keyboards/demiurge/readme.md:3]:
```suggestion
![demiurge-img-1](https://i.imgur.com/gnQ2Pj6h.jpeg)
```

File: keyboards/demiurge/readme.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+# Keyboard Info
+
+- 60% XT
+- Keyboard Maintainer: [ojthetiny](https://github.com/ojthetiny)
+- Hardware Supported: Demiurge PCB
+- Hardware Availability: Through group-buy
+
+Make examples for this keyboard (after setting up your build environment):
```
Comment (fauxpark) [inline keyboards/demiurge/readme.md:12]:
```suggestion
Make example for this keyboard (after setting up your build environment):
```

File: keyboards/demiurge/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS,          KC_TRNS,          KC_TRNS
+	),
+
+	[3] = LAYOUT_all(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS,          KC_TRNS,          KC_TRNS
+	)
```
Comment (fauxpark) [inline keyboards/demiurge/keymaps/default/keymap.c:52]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**